src/runtime_src/core/pcie/common/dmatest.h

```diff
@@ -44,17 +47,23 @@ namespace xcldev {
     };
 
     class DMARunner {
-        std::vector<xclBufferHandle> mBOList;
+        // Ideally I would use a C++ BO object which hides this detail completely inside
+        // but that feature is coming in a future release. For now use a poor man implementation.
+        // DMARunner now uses xclAllocUserPtrBO() to allocate buffers. This reduces memory pressure on
+        // Linux kernel which other wise tries very hard inside xocl to allocate and pin pages when
+        // xlcAllocBO() is used may oops.
+        std::vector<std::pair<xclBufferHandle, xrt_core::aligned_ptr_type>> mBOList;
```

Comment:
You might wish to consider defining a "using-directive" for the "pair" in the vector since this syntax is used in multiple locations below.  Something like:
```
using std::pair<xclBufferHandle, xrt_core::aligned_ptr_type>  buffer_and_deleter;
std::vector<buffer_and_deleter> mBOList;
```

Reply:
Done